spanner/google/cloud/spanner/_helpers.py

```diff
@@ -38,6 +38,7 @@ class TimestampWithNanoseconds(datetime.datetime):
     """
     __slots__ = ('_nanosecond',)
 
+    # pylint: disable=arguments-differ
```

Comment:
I'm not sure:  we are overriding `datetime.datetime.__new__` in Python:  that method is defined in C as a builtin, and cannot be introspected the same way.  I can't think of any other way to acoomplish the goal, either.